src/test/polymer/polymer-core-feature_test.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+import * as path from 'path';
+
+import {Analyzer} from '../../analyzer';
+import {Visitor} from '../../javascript/estree-visitor';
+import {JavaScriptParser} from '../../javascript/javascript-parser';
+import {PolymerCoreFeatureScanner} from '../../polymer/polymer-core-feature-scanner';
+import {FSUrlLoader} from '../../url-loader/fs-url-loader';
+
+suite('PolymerCoreFeatureScanner', () => {
+
+  test('scans _addFeature calls and the Polymer.Base assignment', async() => {
+    const js = `
+      /** Feature A */
+      Polymer.Base._addFeature({
+        /** Method A */
+        methodA: function() {},
+        /** Prop A */
+        propA: []
+      });
+
+      /** Feature B */
+      Polymer.Base._addFeature({
+        methodB: function() {}
+      });
+
+      /** Polymer.Base declaration */
+      Polymer.Base = {
+        methodBase: function() {}
+      };
+
+      /** Invalid feature */
+      Polymer.Base._addFeature(null);
+
+      /** Not a feature at all */
+      Polymer.Base._somethingElse({
+        methodX: function() {}
+      });
+    `;
+
+    const parser = new JavaScriptParser();
+    const scanner = new PolymerCoreFeatureScanner();
+    const doc = parser.parse(js, 'features.js');
+    const visit = (visitor: Visitor) => Promise.resolve(doc.visit([visitor]));
+    const {features} = await scanner.scan(doc, visit);
+
+    assert.lengthOf(features, 4);
+    const [a, b, base, invalid] = features;
+
+    assert.equal(a.description, 'Feature A');
+    assert.lengthOf(a.warnings, 0);
```

Comment:
Opinion: testing lengths is a last resort, because the error message is so very unhelpful. Consider instead:

```typescript
assert.deepEqual(a.warnings, []);
assert.deepEqual(
  a.methods.map({name, type, description} => ({name, type, description})), 
  [{name: 'methodA', type: 'Function', description: 'Method A'}]
);
```

It's shorter, but more importantly, the error message will be actionable because it will give you a diff of what was expected and what was found, which is usually all you need to track down the bug.